invenio_stats/tasks.py

```diff
@@ -100,19 +101,22 @@ def get_bookmark(self):
                                 doc_type='{0}-bookmark'.format(self.event))
         query_bookmark.aggs.metric('latest', 'max', field='date')
         bookmark = query_bookmark.execute()[0]
-        bookmark = datetime.datetime.strptime(bookmark.date, '%Y-%m-%d')
+        # change it to doc_id_suffix
+        bookmark = datetime.datetime.strptime(bookmark.date,
+                                              self.supported_intervals[
+                                                  self.aggregation_interval])
         return bookmark.date()
 
     def set_bookmark(self):
         """Set bookmark for starting next aggregation."""
+        current_search_client.indices.flush(index='*')
+
         def _success_date():
             bookmark = {
-                'date': datetime.date.today().isoformat()
+                'date': datetime.date.today().strftime(
```

Comment:
Right, I changed the bookmark from date to datetime for this. `strftime` works the same way for datetime, and it supports aggregations by hour (to be tested).